intersect.js

```diff
@@ -783,16 +757,39 @@ function curveBBox(x0, y0, x1, y1, x2, y2, x3, y3) {
   };
 }
 
-function pathToCurve(path) {
+/**
+ * Handles caches
+ */
+function getPathCurve(path) {
 
-  var pth = paths(path);
+  const pth = paths(path);
 
   // return cached curve, if existing
   if (pth.curve) {
     return pathClone(pth.curve);
   }
 
-  var curvedPath = pathToAbsolute(path),
+  // retrieve abs path OR create and cache if non existing
+  const abs = pth.abs ||
+    (pth.abs = pathToAbsolute(
+
+      // retrieve parsed path OR create and cache if non existing
+      pth.arr ||
+      (pth.arr = (
+
+        // rough assumption
+        (!isArray(path) || !isArray(path && path[0]))) ?
```

Comment:
I don't understand this...

```suggestion
        typeof path === 'string' ?
```

Reply:
@nikku @barmac can I commit this?